src/main/java/org/commcare/formplayer/services/ScheduledTasks.java

```diff
@@ -25,20 +29,37 @@ public class ScheduledTasks {
     @Autowired
     private FormSessionService formSessionService;
 
+    @Autowired
+    private EntitiesSelectionService entitiesSelectionService;
+
     // the default "0 0 0 * * *" schedule means midnight each night
-    @Scheduled(cron= "${commcare.formplayer.scheduledTasks.purge.cron:0 0 0 * * *}")
+    @Scheduled(cron = "${commcare.formplayer.scheduledTasks.purge.cron:0 0 0 * * *}")
     @SchedulerLock(name = "purge",
             lockAtMostFor = "${commcare.formplayer.scheduledTasks.purge.lockAtMostFor:5h}",
             lockAtLeastFor = "${commcare.formplayer.scheduledTasks.purge.lockAtLeastFor:1h}")
     public void purge() {
         log.info("Starting purge scheduled task.");
-        int deletedRows = formSessionService.purge();
+        doTimedPurge("formSession", formSessionService::purge);
+        doTimedPurge("entitiesSelections", entitiesSelectionService::purge);
+        datadogStatsDClient.increment(
+                String.format("%s.%s", Constants.SCHEDULED_TASKS_PURGE, "timesRun")
+        );
+    }
+
+    private void doTimedPurge(String tag, CheckedFunction<Instant, Integer, RuntimeException> purgeable) {
+        Instant cutoff = Instant.now().minus(7, ChronoUnit.DAYS);
+        log.info("Beginning purge for " + tag);
+        long start = System.currentTimeMillis();
+        int deletedRows = purgeable.apply(cutoff);
+        long elapsed = System.currentTimeMillis() - start;
+        log.info(String.format("Purged %d records in %d ms for %s", deletedRows, elapsed, tag));
         datadogStatsDClient.count(
-                String.format("%s.%s", Constants.SCHEDULED_TASKS_PURGE, "deletedRows"),
+                String.format("%s.%s.%s", Constants.SCHEDULED_TASKS_PURGE, "deletedRows", tag),
                 deletedRows
         );
-        datadogStatsDClient.increment(
-                String.format("%s.%s", Constants.SCHEDULED_TASKS_PURGE, "timesRun")
+        datadogStatsDClient.time(
```

Comment:
this results in change in the Datadog keys from before, is that alright ?

Reply:
I think that's OK